### ISSUE TITLE
Discovery on Time and Attendance barcoding viability and hardware sourcing

### ISSUE THREAD

**Issue by dianamartin** [cityofaustin/atd-data-tech#18288]
# Background
SBO is made up of 9 different work groups spread across 4 separate locations. 
- Greg G. has worked on a timeclock prototype platform 
- He has thought that minimum 9 hardware devices are needed for this project
- Christina has mentioned potentially more if we have multiple entrance/exits and so that we don't have to have a line for signing in for the day. 
 
## Requirements
- Hardware that can connect to wi-fi (so that web platform can be reached)
  - laptops
  - tablet
- USB hardwired scanners: $15-20
- standing scanner holder (similar to grocery stores)
- USB - USB-C adapter or USB - Lightning adapter: $1-12

## Plan
- Is to assigned the hardware devices to the superintendent
- Hardware would be plugged in (near an outlet)
- On a tablet stand?
- Barcode scanner holder

**Comment by dianamartin** [comment]
# 7/16/2024

## Scanner Hardware
On 7/9/2024, found [Knack article re: barcode scanners](https://learn.knack.com/article/ypegi21hbc-can-i-use-a-barcode-or-credit-card-scanner-with-knack) - "**Can I use a barcode or credit card scanner with Knack**?"
Knack test verified bar code scanner:

Name | Description | Notes | Cost 
-- | -- | -- | -- 
Symcode USB 2D Barcode Sanner | [Symcode USB 2D Barcode Scanner Handheld Wired Automatic QR Code Scanner PDF417 Data Matrix Bar Code Scanner with Long USB Cable for Mobile Payment, Convenience Store, Supermarket, Warehouse](https://www.amazon.com/Barcode-Scanner-Symcode-Handheld-Reader/dp/B01LXNM0Y1/ref=sr_1_7?s=office-electronics&ie=UTF8&qid=1507221051&sr=1-7&keywords=barcode+scanner) | Knack verified | $15.19 (Amazon)
Adesso Barcode Scanner 3200U | - [Amazon](https://www.amazon.com/Adesso-NUSCAN-Document-Barcode-Scanner/dp/B001667F28/ref=cm_cr_arp_d_product_top?ie=UTF8) - _Not Available_</br> - [Ebay](https://www.ebay.com/itm/204534808495) | DTS has one on hand | $17.42 (Ebay)
Symcode USB 1D Barcode Scanner | [Symcode USB 1D Barcode Scanner, Handheld Wired CCD Laser Barcode Reader Scanner Supports Screen Scan UPC Bar Code Scanner for Warehouse, Library, Supermarket, POS](https://www.amazon.com/Symcode-Handheld-Supports-Warehouse-Supermarket/dp/B0CSD8FP5W/ref=sr_1_2_sspa?crid=5FOMB7Q3J1BH&dib=eyJ2IjoiMSJ9.tVYTT5TqvUsed_JyDE7ifbVNrTF-z-Fatp6YEa8nLmktehlQLnI-C3Sp5mjigEtmvdCfC8aQQKzQsKIWeqn3dGelwEsCOYUVC1TWpGGQqli1o9xrusk7npSu4XDRt1nraDJcDWRX6xBM3agW0hnA3x5hPYMy3mBSJPv3oFUZ96dP6yWSyeAg9hQHCWsoCV6tBPQq_4Fk1Rm0M0pzeUKp_scHve_t7vvc5OIVRB7Dpt5Egbx_-IrDdaW80h73Ysx8_y0e5PWP1iqPXQ5RPtkyf705ifndwAqcNuZ4AwBUnuc.50tQY-U-08cjufYbY0ep9iQ936ijmSS7_9UJ791O0d0&dib_tag=se&keywords=barcode%2Bscanner%2Busb&qid=1721248576&s=electronics&sprefix=barcode%2Bscanner%2Busb%2Celectronics%2C136&sr=1-2-spons&sp_csd=d2lkZ2V0TmFtZT1zcF9hdGY&th=1) | | $12.99 (Amazon)


### Test 1
- [x] Tested with Dell laptop + [Adesso USB barcode scanner](https://www.ebay.com/itm/204534808495) + HR Knack page "Employee Log"
- Last week, tested barcoding functionality and demoed to stakeholders on 7/10/24
-  

### Test 2
- Showcased during our "Additional Scoping on SBO Time & Attendance Feature" to stakeholders

**Comment by dianamartin** [comment]
# Hardware Options
I need to talk to Tech Services to see what laptop hardware we have (older), but not out of life or needing a refresh

## Option 1
- Laptop (9) - $0 
- Scanners - $136.71
- Adapter - $24.95
- Estimated Total: **$161.66**

## Option 2
- Tablets (6) - free (already have on hand - no purchase)
- Laptop (3) - free (already have on hand - no purchase)
- Scanners - $136.71
- Adapter - $24.95
- Estimated Total: **$161.66**

**Comment by dianamartin** [comment]
# 7/22/2024
- Need to confirm with Greg on the 9 locations and where exactly they will be
- Need to put together PR for it

## Need to Have
- [x] Source barcode scanner
- [x] Source dual splitter 
- [x] Source tablet devices
>tested ipad 10th generation (USB-C), used ipad mini. I think I'm leaning towards ipad minis as the mobile optimized code is compati
- [ ] Source secure kiosk stand

## Nice to Have
- [ ] Barcode scanner holder

**Comment by dianamartin** [comment]
# 7/23/24
- Created a [spreadsheet ](https://cityofaustin.sharepoint.com/sites/tpwDTS/_layouts/15/doc.aspx?sourcedoc={a26e6a7d-f969-432b-aedc-cb3b4841a17d}&action=edit) with all the prices and the proposed photo locations. 
- Christina and I synced and think we need to do field visits to see where the devices will actually be placed. 
- A single set up (tablet, kiosk mount, barcode scanner, adapter) is roughly = `~$535.56`

**Comment by dianamartin** [comment]
# 8/16/24
Purchase Request `PR 2024-8963`, Approved on 8/6/24 
- Ordered items on 8/12/24

Quantity | Item | Notes | Delivery Notes
-- | -- | -- | --
12 | Symcode USB 1DBarcode Scanner, Handheld Wired CCDLaser Barcode ReaderScanner SupportsScreen Scan UPC BarCode Scanner forWarehouse, Library,Supermarket, POS | Have been testing and it's working as expected | Received
10 | USB C Splitter type c toDual USB A FemaleAdapter Type C to DualUSB A double usb 2.0 - Adapter double usb2.0 OTG Splitter CordConverter for MacBookPro, Google Pixel,Galaxy S20 S22 S9 /Note 8 | Tested 8/16/24: Turns out this doesn't charge at the same time as the barcode scanner is plugged in. Might need to get a different one :( | Received, but might return?
10 | iBOLT TabDock Dock'nLock Bizmount - Heavy Duty IndustrialComposite LockingDrill Base Mount for All7" - 10" Tablets- forTrucks/Vehicles/WallMount: Great for ELD,Fleets, and Restaurants | Emailed SBO about installation | Received 
2 | VENREA UniversalBarcode ScannerHolder, BarcodeScanning Holder,Barcode Reader RackBracket Wall MountAccessiors Fit for MostScanner BarcodeScanners Label Readers(5pcs-Style A) | in a 5 pack | Pending Delivery

Emailed SBO Leadership 
CC'd Jennifer, Christina, Peggy
>Hi everyone, 

>I’m writing to let you know that I’m starting to get hardware for the iPads for the Time and Attendance project. I have barcode scanners, splitters, iPads, and the tablet mounts in my possession (still waiting on barcode scanner mounts) but would like to know how we want to go about installing them at the various work sites. I believe I heard that y’all installed the car management lock boxes, would we be doing that same thing?

>Also, I would like to get a list of users (ex. Division Managers, Superintendents..?) that you want to assign the tablets to that will be responsible for being logged into the tablets. I need to get my team named users so that they can configure the tablets. 

>Perhaps we should set up a follow-up meeting to discuss?

>Let me know how you want to proceed.

**Comment by dianamartin** [comment]
# 10/16/2024
Discovery was completed in early August, the hardware ordering came together. Items ordered on 8/12/24.
